src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1301,17 +1264,28 @@ DatabaseShardImp::available() const
 std::unique_ptr<DatabaseShard>
 make_ShardStore(
     Application& app,
-    std::string const& name,
+    Stoppable& parent,
     Scheduler& scheduler,
     int readThreads,
     beast::Journal j)
 {
-    return std::make_unique<DatabaseShardImp>(
+    // The shard store is optional. Future changes will require it.
+    Section const& section {
+        app.config().section(ConfigSection::shardDatabase())};
+    if (section.empty())
+        return nullptr;
+
+    auto shardStore = std::make_unique<DatabaseShardImp>(
         app,
-        name,
+        parent,
+        "ShardStore",
         scheduler,
         readThreads,
         j);
+    if (!shardStore->init())
+        Throw<std::runtime_error>("shard store failed to initialize.");
```

Comment:
Only concern was that this function didn't throw before, but now it could. We could keep the behavior the same by changing it to:
```
if (!shardStore->init()) 
    shardStore.reset();
```

Reply:
Good catch. Init already logged.